src/main/java/pl/javastart/task/UniversityApp.java

```diff
@@ -12,8 +23,36 @@ public class UniversityApp {
      * @param firstName - imię prowadzącego
      * @param lastName  - nazwisko prowadzącego
      */
+
     public void createLecturer(int id, String degree, String firstName, String lastName) {
+        if (!isLecturerExists(id)) {
+            lecturers[lecturersCounter] = new Lecturer(firstName, lastName, id, degree);
+            lecturersCounter++;
+        } else {
+            System.out.println("Prowadzący z id " + id + " już istnieje");
+        }
+    }
+
+    public boolean isLecturerExists(int id) {
+        boolean lecturerExists = false;
+        int i = 0;
+        while (!lecturerExists && i < lecturersCounter) {
+            if (lecturers[i].getId() == id) {
+                lecturerExists = true;
+            }
+            i++;
+        }
+        return lecturerExists;
+    }
 
+    public void printLecturerInfo(int id) {
+        int i = 0;
+        while (lecturers[i].getId() != id) {
+            i++;
+        }
```

Comment:
Logika tutaj sprowadza się do wyszukiwania prowadzącego i powtarza Ci się to choćby z metodą isLecturerExists()
Możesz zdefiniować metodę, która zwraca prowadzącego, albo null i wtedy pętlę do wyszukiwania będziesz miał w jednym miejscu. Coś typu `Lecturer findLecturerById(int id)`